src/folding/nova/nifs.rs

```diff
@@ -214,19 +266,41 @@ mod tests {
         let mut rng = ark_std::test_rng();
         let pedersen_params = Pedersen::<Projective>::new_params(&mut rng, r1cs.A.n_cols);
 
-        let r = Fr::rand(&mut rng); // folding challenge would come from the transcript
+        let poseidon_config = poseidon_test_config::<Fr>();
 
         // compute committed instances
         let ci1 = w1.commit(&pedersen_params, x1.clone());
         let ci2 = w2.commit(&pedersen_params, x2.clone());
 
         // NIFS.P
-        let (w3, _, T, cmT) =
-            NIFS::<Projective>::prove(&pedersen_params, r, &r1cs, &w1, &ci1, &w2, &ci2);
+        let (w3, _, T, cmT) = NIFS::<Projective>::prove(
+            &pedersen_params,
+            poseidon_config.clone(),
+            <Projective as Group>::ScalarField::ZERO,
+            &r1cs,
+            &w1,
+            &ci1,
+            &w2,
+            &ci2,
+        );
 
         // NIFS.V
-        let ci3 = NIFS::<Projective>::verify(r, &ci1, &ci2, &cmT);
+        let ci3 = NIFS::<Projective>::verify(
+            poseidon_config.clone(),
+            <Projective as Group>::ScalarField::ZERO,
+            &ci1,
+            &ci2,
+            &cmT,
+        );
 
+        // compute a challenge
+        let r = compute_challenge::<Projective>(
+            poseidon_config,
+            <Projective as Group>::ScalarField::ZERO,
+            &ci1,
+            &ci2,
+            &cmT,
+        );
```

Comment:
This was done with a random generated `r` instead of a full trancript building & appending, since this is a test and we don't need to recreate the full transcript, which adds complexity to the logic of the test and extra lines to read.